### ISSUE TITLE
ES6: Add tests for updated identifier syntax

### ISSUE THREAD

**Issue by claudepache** [compat-table/compat-table#1026]
Reference:

ES6: http://www.ecma-international.org/ecma-262/6.0/#sec-names-and-keywords

compare with ES5.1: http://www.ecma-international.org/ecma-262/5.1/#sec-7.6

Testcases:

(1) The following program

```js
var ⸯ;
```

(where ⸯ is U+2E2F VERTICAL TILDE) is correctly rejected in Chrome, Edge and Firefox Nightly, but is incorrectly accepted in Safari and Firefox 51.

Webkit bug: https://bugs.webkit.org/show_bug.cgi?id=121541
Mozilla bug: https://bugzilla.mozilla.org/show_bug.cgi?id=917436 (fixed in FF52)

(2) The following program

```js
var 𐋀;
```

(where 𐋀 is U+102C0 CARIAN LETTER G) is correctly accepted in Firefox, Chrome and Edge, but is incorrectly rejected in Safari, .

Curiously, `var \u{102C0};` does work in Safari.

Webkit bug: https://bugs.webkit.org/show_bug.cgi?id=167940

(3) The testcase `var v\u0061r;` (should be rejected) is already included in the compat table (in Miscellaneous > no escaped reserved words as identifiers).

(4) Are there other interesting cases?

**Comment by afmenez** [comment]
I can try to write the tests.  Do you have title suggestions for (1) and (2)?

**Comment by afmenez** [comment]
I am trying to write the tests.  Do you have title suggestions for (1) and (2)?

I have tried to use the same "framework" set by the existing (3) test, but was not successful.
(1) For this one the test is not passing on Firefox 54, Chrome 59 and Edge 14, but according to the description it should (on what versions?).  Any hint?
```
      exec: function() {/*
        var ⸯ;
        try {
          eval('var ⸯ');
        } catch(e) {
          return true;
        }
      */},
```
(2) For this one the test is passing on Firefox 54, Chrome 59 and Edge 14, and it is failing on IE11.  Can assume it is OK?  What are the first version of each browser where the test should pass?
```
      exec: function() {/*
        var 𐋀;
        try {
          eval('var 𐋀');
          return true;
        } catch(e) {
          return false;
        }
      */},
```
(3) I just moved the test from the misc section to the new one.  I assume it is already fine.

**Comment by claudepache** [comment]
> (1)
> exec: function() {/*
> var ⸯ;
> try {
> eval('var ⸯ');
> } catch(e) {
> return true;
> }
> /}

The first `var ⸯ;` must be discarded, since it should throw a SyntaxError. Without that line, the test seems correct.

**Comment by afmenez** [comment]
Indeed.  Can I remove the declaration from the other 2 tests then?
And how was it working on (3) before, shouldn't it throw the same SyntaxError?

**Comment by claudepache** [comment]
For test (2), you could just write:
```js
exec: function() {/*
var 𐋀;
return true;
*/},
```

> And how was it working on (3) before, shouldn't it throw the same SyntaxError?

You didn’t read carefully enough the source code of test (3). Hint:

```js
var \u0061; // legal, equivalent to var a;
var v\u0061r; // illegal, equivalent to var var;
```

**Comment by afmenez** [comment]
Got it.  Thanks for explaining!
Still missing before I can submit the patch:

1. titles for subsections (1) and (2);
2. first version supported for (1) and (2) on Chrome and Edge  (I am using Firefox 52, Chrome 58 and Edge 14 for now, which are the ones I can test);

Anyone?

**Comment by afmenez** [comment]
I believe this was fixed by my patch in #1121 and can be close already.
